migrations/V172__Add_Contract_Numbers_table.sql

```diff
@@ -0,0 +1,14 @@
+CREATE TABLE IF NOT EXISTS system_intake_contract_numbers (
+    id UUID PRIMARY KEY NOT NULL,
+    intake_id UUID REFERENCES system_intakes(id) ON DELETE CASCADE,
+    contract_number ZERO_STRING NOT NULL,
+    created_by TEXT NOT NULL CHECK (created_by ~ '^[A-Z0-9]{4}$'),
```

Comment:
@samoddball, now that the user_account table is implemented, I think that we can and should reference that table instead
`created_by UUID NOT NULL REFERENCES user_account(id),`
` modified_by UUID REFERENCES user_account(id),`

this has some other implications about base struct that is not in the code base at the moment. I stubbed it out when working on POCS for user account, I think that they could be added in here! [Branch](https://github.com/CMSgov/easi-app/compare/main...EASI-3341/user_account_spike_poc).

I'd look at BaseStructUser and the implementation for a TRBRequest. Let me know if you have any questions or would like to pair, the implementation in the poc branch is not completely fleshed out, but we have a well defined paradigm in the MINT application.

Reply:
Thanks again for your help today! This is an awesome feature for the new user account implementation!